pkg/virt-handler/vm.go

```diff
@@ -2561,6 +2564,73 @@ func (d *VirtualMachineController) vmUpdateHelperMigrationTarget(origVMI *v1.Vir
 	}
 	return nil
 }
+
+func ConfigureHousekeepingCgroup(vmi *v1.VirtualMachineInstance, d *VirtualMachineController) error {
+	cgroupManager, err := cgroup.NewManagerFromVM(vmi)
+	if err != nil {
+		return err
+	}
+
+	exists, err := cgroupManager.CreateChildCgroup("housekeeping", "cpuset")
+	// bail out if cgroup already configured
+	if exists == true {
+		return nil
+	}
+	if err != nil {
+		log.Log.Reason(err).Error("CreateChildCgroup ");
+		return err
+	}
+
+	key := controller.VirtualMachineInstanceKey(vmi)
+	domain, _, _, err := d.getDomainFromCache(key)
+	if err != nil {
+		return err
+	}
+
+	hkcpu, err := strconv.Atoi(domain.Spec.CPUTune.EmulatorPin.CPUSet)
+	if err != nil {
+		return err
+	}
+
+
+	log.Log.Object(vmi).Infof("housekeeping cpu: %v", hkcpu)
+
+	err = cgroup.SetCpuSet(cgroupManager, "housekeeping", []int{hkcpu})
+	if err != nil {
+		return err
+	}
+
+	tids, err := cgroup.GetCgroupThreads(cgroupManager)
+	if err != nil {
+		return err
+	}
+	hktids := make([]int, 0, 10)
+
+	for _, tid := range tids {
+		proc, err := ps.FindProcess(tid)
+		if err != nil {
+			log.Log.Object(vmi).Errorf("Failure to find process: %s", err.Error())
+			return err
+		}
+		comm := proc.Executable()
+		if strings.Contains(comm, "CPU ") && strings.Contains(comm, "KVM") {
```

Comment:
is the extra space in CPU (`"CPU "`) relevant or it does not matter?

Reply:
QEMU sets the thread name as "CPU X/KVM", so that is what is being compared (the space is intentional for a more strict match). Thanks.

Reply:
Will this work for emulation?

Reply:
Good point! I see `CPU 0/TCG`